cas-server-webapp-actions-aup-ldap/src/main/java/org/jasig/cas/web/flow/LdapAcceptableUsagePolicyRepository.java

```diff
@@ -0,0 +1,177 @@
+import org.jasig.cas.authentication.Credential;
+import org.ldaptive.AttributeModification;
+import org.ldaptive.AttributeModificationType;
+import org.ldaptive.Connection;
+import org.ldaptive.ConnectionFactory;
+import org.ldaptive.LdapAttribute;
+import org.ldaptive.LdapEntry;
+import org.ldaptive.LdapException;
+import org.ldaptive.ModifyOperation;
+import org.ldaptive.ModifyRequest;
+import org.ldaptive.Response;
+import org.ldaptive.ReturnAttributes;
+import org.ldaptive.SearchFilter;
+import org.ldaptive.SearchOperation;
+import org.ldaptive.SearchRequest;
+import org.ldaptive.SearchResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.webflow.execution.RequestContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is {@link LdapAcceptableUsagePolicyRepository}.
+ * Examines the principal attribute collection to determine if
+ * the policy has been accepted, and if not, allows for a configurable
+ * way so that user's choice can later be remembered and saved back into
+ * the LDAP instance.
+ *
+ * @author Misagh Moayyed
+ * @since 4.2
+ */
+@Component("ldapAcceptableUsagePolicyRepository")
+public class LdapAcceptableUsagePolicyRepository extends AbstractPrincipalAttributeAcceptableUsagePolicyRepository {
+    private static final long serialVersionUID = 1600024683199961892L;
+
+    @Autowired
+    @Qualifier("ldapUsagePolicyConnectionFactory")
+    private ConnectionFactory connectionFactory;
+
+    @Autowired
+    @Qualifier("ldapUsagePolicySearchRequest")
+    private SearchRequest searchRequest;
+
+    @Value("${cas.aup.ldap.search.filter:}")
+    private String searchFilter;
+
+    @Override
+    public boolean submit(final RequestContext requestContext, final Credential credential) {
+
+        String currentDn = null;
+        try (final Connection searchConnection = getConnection()) {
+            final Response<SearchResult> response = searchForId(searchConnection, credential.getId());
+            if (hasResults(response)) {
+                currentDn = response.getResult().getEntry().getDn();
+            }
+        } catch (final Exception e) {
+            logger.error(e.getMessage(), e);
+        }
+
+        if (StringUtils.isNotBlank(currentDn)) {
+            logger.debug("Updating {}", currentDn);
+
+            try (final Connection modifyConnection = getConnection()) {
+                final ModifyOperation operation = new ModifyOperation(modifyConnection);
+                final List<AttributeModification> mods = new ArrayList<>();
+
+                final LdapEntry entry = new LdapEntry(currentDn, new LdapAttribute(this.aupAttributeName,
+                        Boolean.TRUE.toString()));
+                for (final LdapAttribute attr : entry.getAttributes()) {
```

Comment:
OK, in the spirit of being consistent with functional style through the code base, this imperative style could be refactored with functional style like this:

```java
AttributeModification[] mods = 
                        entry.getAttributes()
                        .stream()
                        .map(attr -> new AttributeModification(AttributeModificationType.REPLACE, attr))
                        .toArray(size -> new AttributeModification[size]);
                
                final ModifyRequest request = new ModifyRequest(currentDn, mods);
``` 

I do realize this is for `4.2.x`, but just to have a record here when this will be going into the master...